pkg/connector/helpers.go

```diff
@@ -60,3 +61,24 @@ func getPageTokenFromOffset(bag *pagination.Bag, offset int64) (string, error) {
 
 	return pageToken, nil
 }
+
+// Unfortunatelly, the Jira API does not provide a way to get the role id from project.
+// It only provides a link to the role. Like this: https://your-domain.atlassian.net/rest/api/3/project/10001/role/10002
+// So, we need to parse the role id from the link.
+func parseRoleIdFromRoleLink(roleLink string) (int, error) {
+	regexPattern := `\/(\d+)\/?$` // Regex pattern to match the last number in the URL path
```

Comment:
Instead of using a regex, this function could parse the url, then split the path on / and return the sixth element. It could also check that the relevant elements match (eg: pathElems[3] is "project" and pathElems[5] is "role")

I realize you didn't modify this function, but the existing function will break if the link has query args after the path. eg: `https://your-domain.atlassian.net/rest/api/3/project/10001/role/10002?blah=1234` will fail.